agimus_controller/agimus_controller/ocp/ocp_traj_tracking_collision_avoidance.py

```diff
@@ -0,0 +1,244 @@
+            self._state, self._actuation, terminal_cost_model, constraints
+        )
+
+        terminal_model = crocoddyl.IntegratedActionModelEuler(terminal_DAM, 0.0)
+        terminal_model.differential.armature = self._robot_models.armature
+        return terminal_model
+
+    def get_collision_constraints(self, active=True):
+        """Returns the collision constraints model manager."""
+        constraint_model_manager = crocoddyl.ConstraintModelManager(
+            self._state, self.nq
+        )
+        if len(self._collision_model.collisionPairs) != 0:
+            for col_idx in range(len(self._collision_model.collisionPairs)):
+                collision_constraint = self._get_collision_constraint_residual(col_idx)
+                # Adding the constraint to the constraint manager
+                constraint_model_manager.addConstraint(
+                    "col_term_" + str(col_idx), collision_constraint, active=active
+                )
+        return constraint_model_manager
+
+    def _get_collision_constraint_residual(
+        self, col_idx: int
+    ) -> "crocoddyl.ConstraintModelResidual":
+        """Returns the collision constraint model residual."""
+        distance_coll_residual = ResidualDistanceCollision(
+            self._state, 7, self._collision_model, col_idx
+        )
+        return crocoddyl.ConstraintModelResidual(
+            self._state,
+            distance_coll_residual,
+            np.array([self._params.collision_safety_margin]),
+            np.array([np.inf]),
+        )
+
+    def set_reference_weighted_trajectory(
+        self, reference_weighted_trajectory: list[WeightedTrajectoryPoint]
+    ):
+        """Set the reference trajectory for the OCP."""
+        super().set_reference_weighted_trajectory(reference_weighted_trajectory)
+        # Modify running costs reference and weights
+        for i in range(self.horizon_size - 1):
+            model = self._solver.problem.runningModels[i]
+
+            state_ref = np.concatenate(
+                (
+                    reference_weighted_trajectory[i].point.robot_configuration,
+                    reference_weighted_trajectory[i].point.robot_velocity,
+                )
+            )
+            state_weights = np.concatenate(
+                (
+                    reference_weighted_trajectory[i].weights.w_robot_configuration,
+                    reference_weighted_trajectory[i].weights.w_robot_velocity,
+                )
+            )
+            self.modify_cost_reference_and_weights(
+                model, "stateReg", state_ref, state_weights
+            )
+
+            u_ref = reference_weighted_trajectory[i].point.robot_effort
+            u_weights = reference_weighted_trajectory[i].weights.w_robot_effort
+            self.modify_cost_reference_and_weights(model, "ctrlReg", u_ref, u_weights)
+
+            ee_names = list(
+                iter(reference_weighted_trajectory[i].weights.w_end_effector_poses)
+            )
```

Comment:
```suggestion
            reference_weighted_trajectory[i].weights.w_end_effector_poses.keys()
```
This should be equivalent